router/throttler/throttler.go

```diff
@@ -0,0 +1,166 @@
+		},
+		// https://help.amplitude.com/hc/en-us/articles/360032842391-HTTP-API-V2#upload-limit
+		"AM": {
+			"limit":                     1000,
+			"timeWindowInS":             1,
+			"userEventOrderingRequired": true,
+			"userLevelThrottling":       true,
+			"userLevelLimit":            10,
+			"userLevelTimeWindowInS":    1,
+		},
+	}
+}
+
+func (throttler *HandleT) setLimits() {
+	// set eventLimit
+	defaultLimit, _ := parameters[throttler.destination]["limit"].(int)
+	throttler.eventLimit = config.GetInt(fmt.Sprintf(`Router.throttler.%s.limit`, throttler.destination), defaultLimit)
+
+	// set timeWindow
+	defaultTimeWindow, _ := parameters[throttler.destination]["timeWindowInS"].(int)
+	throttler.timeWindow = config.GetDuration(fmt.Sprintf(`Router.throttler.%s.timeWindowInS`, throttler.destination), time.Duration(defaultTimeWindow)) * time.Second
+
+	// set userEventOrderingRequired
+	defautOrderingRequired, ok := parameters[throttler.destination]["userEventOrderingRequired"].(bool)
+	if !ok {
+		defautOrderingRequired = true
+	}
+	throttler.UserEventOrderingRequired = config.GetBool(fmt.Sprintf(`Router.throttler.%s.userEventOrderingRequired`, throttler.destination), defautOrderingRequired)
+
+	// enable throttler
+	if throttler.eventLimit != 0 && throttler.timeWindow != 0 {
+		throttler.permit = true
+	}
+
+	defautUserLevelThrottling, _ := parameters[throttler.destination]["userLevelThrottling"].(bool)
+	throttler.userLevelThrottling = config.GetBool(fmt.Sprintf(`Router.throttler.%s.userLevelThrottling`, throttler.destination), defautUserLevelThrottling)
+
+	if throttler.userLevelThrottling {
+		// set eventLimit for userLevelThrottler
+		userLevelDefaultLimit, _ := parameters[throttler.destination]["userLevelLimit"].(int)
+		throttler.userLevelThrottler.eventLimit = config.GetInt(fmt.Sprintf(`Router.throttler.%s.userLevelLimit`, throttler.destination), userLevelDefaultLimit)
+
+		// set timeWindow for userLevelThrottler
+		userLevelDefaultTimeWindow, _ := parameters[throttler.destination]["userLevelTimeWindowInS"].(int)
+		throttler.userLevelThrottler.timeWindow = config.GetDuration(fmt.Sprintf(`Router.throttler.%s.userLevelTimeWindowInS`, throttler.destination), time.Duration(userLevelDefaultTimeWindow)) * time.Second
+
+		// enable userLevelthrottler
+		if throttler.userLevelThrottler.eventLimit != 0 && throttler.userLevelThrottler.timeWindow != 0 {
+			throttler.userLevelThrottler.permit = true
+		}
+	}
+
+	// TOOD: should buckets be configurable
+	throttler.bucketsInWindow = config.GetInt(fmt.Sprintf(`Router.throttle.%s.bucketsInWindow`, throttler.destination), 12)
+	throttler.userLevelThrottler.bucketsInWindow = config.GetInt(fmt.Sprintf(`Router.throttle.%s.bucketsInWindow`, throttler.destination), 12)
+}
+
+//SetUp eventLimiter
+func (throttler *HandleT) SetUp(destination string) {
+	var userLevelThrottler UserHandleT
+	throttler.userLevelThrottler = &userLevelThrottler
+
+	throttler.destination = destination
+
```

Comment:
Even before setting the limits, let's have a check if throttling can be enabled for this destination and then to return from here? 

if parameters[destinationID] does not exist, return.